pkg/api/validation/validation.go

```diff
@@ -687,11 +687,10 @@ func ValidatePodStatusUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
 
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta).Prefix("metadata")...)
 
-	// TODO: allow change when bindings are properly decoupled from pods
```

Comment:
For consistency, and to catch bugs.

For a few select readonly fields we don't expect the client to know, such as creationTimestamp, we do allow them to be empty. However, in general, PUT doesn't merge. It overwrites. Host is a mutable field, it just can't be changed once written. An attempt to do so, including leaving it empty, is a bug.

We could potentially remove this field from Status in v1beta3, since binding is now writing Spec, but kubelet needs to know its identity in order to update NodeStatus, so it doesn't seem like it should be hard for it to populate this field.

Furthermore, kubelet has to watch and cache the pods in order to know what to run, anyway, so it could do what I proposed for NodeStatus, which is to merge the status updates with the status received from apiserver.